influxdb_iox/tests/query_tests2/setups.rs

```diff
@@ -931,5 +935,96 @@ pub static SETUPS: Lazy<HashMap<SetupName, SetupSteps>> = Lazy::new(|| {
                 },
             ],
         ),
+        (
+            "ThreeChunksWithRetention",
+            vec![
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_partially_inside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_fully_inside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::RecordNumParquetFiles,
+                Step::WriteLineProtocol(RETENTION_SETUP.lp_fully_outside.clone()),
+                Step::Persist,
+                Step::WaitForPersisted2 {
+                    expected_increase: 1,
+                },
+                Step::SetRetention(Some(RETENTION_SETUP.retention_period_ns)),
+            ],
+        ),
     ])
 });
+
+/// Holds parameters for retention period. Tests based on this need to
+/// be run within the hour of being created.
+///
+/// ```text                  (cut off)                    (now)
+/// time: -----------------------|--------------------------|>
+///
+/// partially_inside:   |-----------------|
+///
+/// fully_inside:                 +1h |-----------|
+///
+/// fully_outside: |---------|
+/// ```
+struct RetentionSetup {
+    /// the retention period, relative to now() that the three data
+    /// chunks fall inside/outside
+    retention_period_ns: i64,
+
+    /// lineprotocol data partially inside retention
+    lp_partially_inside: String,
+
+    /// lineprotocol data fully inside (included in query)
+    lp_fully_inside: String,
+
+    /// lineprotocol data fully outside (excluded from query)
+    lp_fully_outside: String,
+}
+
+impl RetentionSetup {
+    fn new() -> Self {
+        let retention_period_1_hour_ns = 3600 * 1_000_000_000;
+
+        // Data is relative to this particular time stamp
+        let cutoff = Time::from_rfc3339("2022-01-01T00:00:00+00:00")
+            .unwrap()
+            .timestamp_nanos();
+        let inside_retention = cutoff + retention_period_1_hour_ns; // 1 hour before it is outside retention
```

Comment:
My brain is struggling with the `+` and the word "before"... I *think* I understand after staring at it, but could we change this to (if I'm understanding correctly):

```suggestion
        // Timestamp 1 hour later than the cutoff, so the data will be retained for 1 hour
        let inside_retention = cutoff + retention_period_1_hour_ns;
```

Reply:
Thank you -- that is a better phrasing